### PR TITLE
MM-25007: Add staticcheck

### PR DESCRIPTION
**Pull request by agnivade**
We ignore test files for now. We can add support gradually.

#### Ticket link

https://mattermost.atlassian.net/browse/MM-25007

### CODE REVIEW

File: app/audit.go [hunk 1]
```diff
@@ -136,7 +136,6 @@ func (s *Server) configureAudit(adt *audit.Audit) {
 		filter := adt.MakeFilter(RestLevel, RestContentLevel, RestPermsLevel, CLILevel)
 		formatter := adt.MakeJSONFormatter()
 		formatter.DisableTimestamp = false
-		formatter.Indent = "\n"
```
Comment (wiggin77) [inline app/audit.go:139]: This is indeed deprecated.

File: .golangci.yml [hunk 2]
```diff
@@ -41,6 +42,10 @@ issues:
       # ignore warnings from table tests. https://github.com/kyoh86/scopelint/issues/4
       path: ".*_test.go|store/storetest"
 
+    - linters:
+      - staticcheck
+      path: ".*_test.go|store/storetest"
```
Comment (hanzei) [inline .golangci.yml:47]: I would love to see a couple of HW tickets to add the checks for test files.

File: i18n/en.json [hunk 3]
```diff
@@ -3866,6 +3866,10 @@
     "id": "ent.compliance.csv.metadata.json.marshalling.appError",
     "translation": "Unable to convert metadata to json."
   },
+  {
+    "id": "ent.compliance.csv.metadata.json.zipfile.appError",
+    "translation": "Unable to create the zip file"
```
Comment (hanzei) [inline i18n/en.json:3871]: I thought CI should fail if a PR missed a translation string. Any idea why `master` is not broken?

Comment (agnivade) [reply i18n/en.json:3871]: This is from the EE PR for staticcheck. I added a new translation there.

File: .golangci.yml (hunk 2 above)
Comment (agnivade) [reply .golangci.yml:47]: Totally agree. Let me see if I can spin up some of these.

**Comment by agnivade** [comment]
@hanzei - My apologies if you expected to review the EE PR too. Please do feel free: https://github.com/mattermost/enterprise/pull/644. And review any one reviewer that you dislike ;)

**Comment by hanzei** [comment]
@agnivade No problem, it just made sense to me to review all related changes.

**Comment by agnivade** [comment]
@lindy65 - It might be hard to QA this as this changes a few lines at various random places throughout the code. And there's nothing really major here. Just some very small changes. Maybe just do a brief sanity test post-merge ? Or feel free to test anything specific that you like.

**Comment by lindy65** [comment]
Thanks @agnivade 

> changes a few lines at various random places throughout the code

Can you tell me which areas of the app this might affect so I can include them in my testing?

**Comment by agnivade** [comment]
Sure. It's all over the codebase though. I can tell you the areas of the codebase, but I don't know what steps are needed from the UI to trigger them. That's why it may be hard to test.

- CreateProfileImage - just the dummy image which gets created.
- Get file info for a post 
- Upload emoji gif greater than 128x128
- Configure audit layer

I don't know what are the expected responses of this, or even how to test this. :( Just that the code touches these parts.

**Pull request merged**